src/status/chat.nim

```diff
@@ -127,6 +129,30 @@ proc updateContacts*(self: ChatModel, contacts: seq[Profile]) =
 
 proc init*(self: ChatModel, pubKey: string) =
   var chatList = status_chat.loadChats()
+  var contacts = getAddedContacts()
+
+  let profileUpdatesChatIds = chatList.filter(c => c.chatType == ChatType.Profile).map(c => c.id)
+
+  if chatList.filter(c => c.chatType == ChatType.Timeline).len == 0:
+    var timelineChannel = newChat(status_utils.getTimelineChatId(), ChatType.Timeline)
+    self.join(timelineChannel.id, timelineChannel.chatType)
+    chatList.add(timelineChannel)
+
+  if not profileUpdatesChatIds.contains("@" & pubKey):
+    var profileUpdateChannel = newChat("@" & pubKey, ChatType.Profile)
+    status_chat.saveChat(profileUpdateChannel.id, profileUpdateChannel.chatType, true, "#000000", "", pubKey)
+    chatList.add(profileUpdateChannel)
+
+  # For profile updates and timeline, we have to make sure that for
+  # each added contact, a chat has been saved for the currently logged-in
+  # user. Users that will use a version of Status with timeline support for the
+  # first time, won't have any of those otherwise.
+  if profileUpdatesChatIds.filter(id => id != "@" & pubKey).len != contacts.len:
+    for contact in contacts:
+      if not profileUpdatesChatIds.contains("@" & contact.address):
+        let profileUpdatesChannel = newChat("@" & contact.address, ChatType.Profile)
+        status_chat.saveChat(profileUpdatesChannel.id, profileUpdatesChannel.chatType, true, "#000000", contact.ensName, contact.address)
```

Comment:
if you implement the saveChat* suggested change:
status_chat.saveChat(profileUpdatesChannel.id, profileUpdatesChannel.chatType, ensName=contact.ensName, profile=contact.address)